### PR TITLE
Adding timeout ability to MYSQL connection

### PR DESCRIPTION
**Pull request by TheSnowFox**

### CODE REVIEW

File: Sources/SwiftKueryMySQL/MySQLConnection.swift [hunk 1]
```diff
@@ -130,6 +133,7 @@ public class MySQLConnection: Connection {
 
         let connectionGenerator: () -> Connection? = {
             let connection = self.init(host: host, user: user, password: password, database: database, port: port, unixSocket: unixSocket, clientFlag: clientFlag, characterSet: characterSet, reconnect: reconnect)
```
Comment (ianpartridge) [inline Sources/SwiftKueryMySQL/MySQLConnection.swift]: Would it be better to pass the timeout through as an `init()` parameter instead?

File: Sources/SwiftKueryMySQL/MySQLConnection.swift [hunk 2]
```diff
@@ -45,6 +45,9 @@ public class MySQLConnection: Connection {
     private let clientFlag: UInt
     private let characterSet: String
     private let reconnect: Bool
+    
+    /// Connection timeout in seconds
```
Comment (ianpartridge) [inline Sources/SwiftKueryMySQL/MySQLConnection.swift]: Why not keep this as milliseconds and do the conversion in `connect()`?

File: Sources/SwiftKueryMySQL/MySQLConnection.swift (hunk 1 above)
Comment (TheSnowFox) [reply Sources/SwiftKueryMySQL/MySQLConnection.swift]: I have some concerns about compatibility for people who already implemented the lib.
_ I can simply modify the current (designated) `required init()` but if anyone subclassed `MySQLConnection`, they would have compile time error when updating their project.
_ I can make a new designated initializer (as required) so the current one become convenience? People will also have compile time error ?
_ I can make a new convenience initializer with timeout parameter but a convenience initializer would have more parameters than the designated? (Sounds strange to me)

That's why I thought not changing the current `init()` preferable.

Again, I'm not an expert so tell me what sounds the best for you, and I will try.

File: Sources/SwiftKueryMySQL/MySQLConnection.swift (hunk 2 above)
Comment (TheSnowFox) [reply Sources/SwiftKueryMySQL/MySQLConnection.swift]: OK. I Will fix it.

File: Sources/SwiftKueryMySQL/MySQLConnection.swift (hunk 1 above)
Comment (ianpartridge) [reply Sources/SwiftKueryMySQL/MySQLConnection.swift]: These are good points, let's leave this as it is then :)

**Comment by TheSnowFox** [comment]
Is it bugged ? The CI test passed 2 days ago but status not updated...

**Comment by ianpartridge** [comment]
@tunniclm are you ok with this change?

**Comment by TheSnowFox** [comment]
Sorry to re-open this thread. Anything new ?

**Comment by EnriqueL8** [comment]
Tests are passing on Travis, must be a bug that the check is still pending....will restart tests.

**Comment by EnriqueL8** [comment]
Tests pass, could we merge this? @ianpartridge @tunniclm

**Comment by tunniclm** [comment]
I only took a quick look over this, but it looks okay to me.

**Pull request merged**